src/main/java/com/enonic/app/guillotine/graphql/transformer/ExtensionGraphQLTypeVisitor.java

```diff
@@ -0,0 +1,139 @@
+{
+
+    private final Map<String, ScriptValue> creationCallbacksDefs;
+
+    public ExtensionGraphQLTypeVisitor( final SchemaExtensions schemaExtensions )
+    {
+        this.creationCallbacksDefs = schemaExtensions.getCreationCallbacks();
+    }
+
+    @Override
+    public TraversalControl visitGraphQLInterfaceType( final GraphQLInterfaceType node,
+                                                       final TraverserContext<GraphQLSchemaElement> context )
+    {
+        if ( creationCallbacksDefs != null && creationCallbacksDefs.get( node.getName() ) != null )
+        {
+            ScriptValue creationCallbackFn = creationCallbacksDefs.get( node.getName() );
+
+            BaseCreationCallbackParams creationCallbackParams = new BaseCreationCallbackParams();
+            creationCallbackFn.call( creationCallbackParams );
+
+            String newDescription = Objects.requireNonNullElse( creationCallbackParams.getDescription(), node.getDescription() );
+            List<GraphQLFieldDefinition> newFields = GraphQLFieldsMerger.merge( node.getFieldDefinitions(), creationCallbackParams );
+
+            GraphQLInterfaceType transformedNode = node.transform( builder -> builder.description( newDescription ).fields( newFields ) );
+
+            return changeNode( context, transformedNode );
+
+        }
+        return super.visitGraphQLInterfaceType( node, context );
+    }
+
+    @Override
+    public TraversalControl visitGraphQLObjectType( final GraphQLObjectType node, final TraverserContext<GraphQLSchemaElement> context )
+    {
+        if ( creationCallbacksDefs != null )
+        {
+            GraphQLObjectType transformedNode = node;
+
+            int transformedCount = 0;
+
+            if ( creationCallbacksDefs.get( transformedNode.getName() ) != null )
+            {
+                ScriptValue creationCallback = creationCallbacksDefs.get( transformedNode.getName() );
+                transformedNode = executeCreationCallback( transformedNode, creationCallback );
+
+                transformedCount++;
+            }
+
+            Set<String> interfaceNames =
+                transformedNode.getInterfaces().stream().map( GraphQLNamedOutputType::getName ).collect( Collectors.toSet() );
+
+            for ( String interfaceName : interfaceNames )
+            {
+                ScriptValue creationCallback = creationCallbacksDefs.get( interfaceName );
+                if ( creationCallback != null )
+                {
+                    BaseCreationCallbackParams creationCallbackParams = new BaseCreationCallbackParams();
+                    creationCallback.call( creationCallbackParams );
+
+                    List<GraphQLFieldDefinition> newFields =
+                        GraphQLFieldsMerger.merge( transformedNode.getFieldDefinitions(), creationCallbackParams );
+
+                    transformedNode = transformedNode.transform( builder -> builder.replaceFields( newFields ) );
+
+                    transformedCount++;
```

Comment:
do you need to count? why not just  transformed = true

Reply:
You are right.